osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecurityKeyService.java

```diff
@@ -1,30 +1,101 @@
+     * communication key that works is known), and you are sure any existing new
+     * key data that is not activated yet (for instance a new key stored earlier in an
+     * attempt to replace the communication key that got aborted).<br>
+     * <p>
+     * The moment the new key is known to be transferred to the device, make
+     * sure to activate it by calling
+     * {@link #activateNewKey(String, SecurityKeyType)}.
+     *
+     * @param deviceIdentification
+     *         DLMS device id
+     * @param key
+     *        key to store, unencrypted
+     * @param keyType
+     *         type of key
+     *
+     *
+     * @see #activateNewKey(String, SecurityKeyType)
+     */
+    void storeNewKey(final String deviceIdentification, final SecurityKeyType keyType, final byte[] key);
+    void storeNewKeys(final String deviceIdentification, final Map<SecurityKeyType, byte[]> keysByType);
+
+    /**
+     * @see #storeNewKey(String, SecurityKeyType, byte[])
+     */
+    //void aesDecryptAndStoreNewKey(final String deviceIdentification, final SecurityKeyType keyTypes,
+    //        final byte[] aesEncryptedKeys) throws FunctionalException;
+
+    boolean hasNewSecretOfType(String deviceIdentification, SecurityKeyType keyType);
+
+    /**
+     * Updates the state of a new key from 'new' to 'active'
+     * <p>
+     * This method should be called to activate a new key stored with
+     * {@link #storeNewKeys(String, Map)} after it has
+     * been confirmed to be set on the device.
+     *
+     * @param deviceIdentification
+     *         DLMS device id
+     * @param keyType
+     *         type of key
+     *
+     * @throws ProtocolAdapterException
+     *         if no new key is stored with the given device
+     * @see #storeNewKeys(String, Map)
+     */
+    void activateNewKey(final String deviceIdentification, final SecurityKeyType keyType)
+            throws ProtocolAdapterException;
+    void activateNewKeys(final String deviceIdentification, final List<SecurityKeyType> keyTypes)
+            throws ProtocolAdapterException;
+
+    byte[] generate128BitsKeyAndStoreAsNewKey(final String deviceIdentification, final SecurityKeyType keyType);
+    Map<SecurityKeyType, byte[]> generate128BitsKeysAndStoreAsNewKeys(final String deviceIdentification,
+            final List<SecurityKeyType> keyTypes);
+
+    //RSA decrypt key (from incoming requests) and encrypt with AES (for in memory storage)
+    //byte[] reEncryptKey(final byte[] externallyEncryptedKey) throws
+    //        FunctionalException;
+
+    //decrypt OSGP provided RSA key
+    byte[] rsaDecrypt(final byte[] externallyEncryptedKey) throws FunctionalException;
+
+    //AES decrypt (decrypt memory storage for actual use of key)
+    //byte[] aesDecryptKey(final byte[] encryptedKey) throws FunctionalException;
+
+    //AES encrypt (encrypt for safe memory storage)
+    //byte[] aesEncryptKey(final byte[] plainKey) throws FunctionalException;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)